internal/terraform/parser/module.go

```diff
@@ -32,7 +32,9 @@ func ParseModuleFiles(fs FS, modPath string) (ast.ModFiles, ast.ModDiags, error)
 
 		src, err := fs.ReadFile(fullPath)
 		if err != nil {
-			return nil, nil, err
+			// If a file isn't accessible, continue with reading the
+			// remaining module files
+			continue
```

Comment:
This makes sense for the edge case described in https://github.com/hashicorp/terraform-ls/issues/1067 but have we considered what impact this would have on real `*.tf` files which aren't accessible, e.g. due to mis-set permissions?

I'm guessing that we are just hoping that both client and server would be aligned on the "readability" of the same file, i.e. they'd be launched under the same user and have the same level of access. i.e. if we can't index the file, it also won't come through `textDocument/didOpen`, right?

So probably nothing to do here - just thinking out loud about the consequences and making sure we're aligned.

One edge case which comes to mind is - what if the user changes permissions at runtime - I'm guessing that this would come through the watch event, right? And presumably it comes before `didOpen` event, so there should be no race condition.

Reply:
That's a valid concern, thanks for raising it. I did some testing with mis-set permissions, but I can't create a test case. It seems impossible commit such a file, because git can't access it 😅 

<img width="733" alt="CleanShot 2023-02-13 at 12 02 10@2x" src="https://user-images.githubusercontent.com/45985/218441182-731999b0-0a06-44f4-9501-a450751e0ca1.png">

The LS will skip it, and VS Code can't open the file, so it shouldn't come through `textDocument/didOpen`.
<img width="263" alt="CleanShot 2023-02-13 at 11 56 10@2x" src="https://user-images.githubusercontent.com/45985/218441276-aff839ea-7044-44d3-bb1c-b818ca5c9d91.png">

If the file permissions change, the watcher will pick it up and the LS will rerun all module related jobs:
<img width="1102" alt="CleanShot 2023-02-13 at 12 06 12@2x" src="https://user-images.githubusercontent.com/45985/218441818-61b7e1b7-8a33-429f-ba2e-3c16df3fc942.png">